thirdparty/misc/pcg.cpp

```diff
@@ -3,13 +3,16 @@
 
 #include "pcg.h"
 
-uint32_t pcg32_random_r(pcg32_random_t* rng)
-{
-    uint64_t oldstate = rng->state;
-    // Advance internal state
-    rng->state = oldstate * 6364136223846793005ULL + (rng->inc|1);
-    // Calculate output function (XSH RR), uses old state for max ILP
-    uint32_t xorshifted = ((oldstate >> 18u) ^ oldstate) >> 27u;
-    uint32_t rot = oldstate >> 59u;
-    return (xorshifted >> rot) | (xorshifted << ((-rot) & 31));
+void pcg32_random_init(pcg32_random_t *rng) {
```

Comment:
This is thirdparty code, so I would prefer if we don't change it.

If we do need to change it, we should at least not change the code style of the thirdparty code, as it generates a bigger diff than necessary. And the changes we made should be clearly identified with `// -- GODOT -- start` and `// -- GODOT -- end` comments.

Reply:
Yeah, it was my fault, I didnt pay attention to thirdparty tag. Well I guess generating one number leads to the same result(with a little performance overhead).